synapse/models/economic.py

```diff
@@ -296,5 +339,125 @@ def getModelDefs(self):
+
+                    ('bank', ('ou:org', {}}), {
+                        'doc': 'The bank which was issued the ABA RTN.'}),
+
+                    ('bank:name', ('ou:name', {}), {
+                        'doc': 'The name which is registered for this ABA RTN.'}),
+
+                )),
+
+                ('econ:bank:iban', {}, ()),
+
+                ('econ:bank:swift:bic', {}, (
+
+                    ('business', ('ou:org', {}), {
+                        'doc': 'The business which is the registered owner of the SWIFT BIC.'}),
+
+                    ('office', ('ps:contact', {}), {
+                        'doc': 'The branch or office which is specified in the last 3 digits of the SWIFT BIC.'}),
+                )),
+
+                ('econ:bank:account:type:taxonomy', {}, ()),
+                ('econ:bank:account', {}, (
+
+                    ('type', ('econ:bank:account:type:taxonomy', {}), {
+                        'doc': 'The type of bank account.'}),
+
+                    ('aba:rtn', ('econ:bank:aba:rtn', {}), {
+                        'doc': 'The ABA routing transit number for the bank which issued the account.'}),
+
+                    ('number', ('str', {'regex': '[0-9]+'}), {
+                        'doc': 'The account number.'}),
+
+                    ('iban', ('econ:bank:iban', {}), {
+                        'doc': 'The IBAN for the account.'}),
+
+                    ('owner', ('ps:contact', {}), {
+                        'doc': 'The contact information associated with the account.'}),
+
+                    ('issuer', ('ou:org', {}), {
+                        'doc': 'The bank which issued the account.'}),
+
+                    ('issuer:name', ('ou:name', {}), {
+                        'doc': 'The name of the bank which issued the account.'}),
+
+                    ('currency', ('econ:currency', {}), {
+                        'doc': 'The currency of the account balance.'}),
+
+                    ('balance', ('econ:bank:balance', {}), {
+                        'doc': 'The most recently known bank balance information.'}),
+                )),
+
+                ('econ:bank:balance', {}, (
+
+                    ('time', ('time', {}), {
+                        'doc': 'The time that the account balance was observed.'}),
+
+                    ('ammount', ('econ:price', {}), {
+                        'doc': 'The amount of currency available at the time.'}),
+
+                    ('account', ('econ:bank:account', {}), {
+                        'doc': 'The bank account which contained the balance amount.'}),
+                )),
+                ('econ:bank:statement', {}, (
+
+                    ('account', ('econ:bank:account', {}), {
+                        'doc': 'The bank account used to compute the statement.'}),
+
+                    ('period', ('ival', {}), {
+                        'doc': 'The period that the statement includes.'}),
+
+                    ('balance', ('econ:price', {}), {
```

Comment:
Since the statement has it's own period, the balance doesn't need the extra layer of indirection. You make a good point about opening / closing balance. I will add :+1: